src/main/java/reactor/kafka/sender/KafkaSender.java

```diff
@@ -44,10 +44,22 @@
      *        after the sender is created will not be used by the sender.
      * @return new instance of Kafka sender
      */
-    public static <K, V> KafkaSender<K, V> create(SenderOptions<K, V> options) {
+    static <K, V> KafkaSender<K, V> create(SenderOptions<K, V> options) {
         return new DefaultKafkaSender<>(ProducerFactory.INSTANCE, options);
     }
 
+    /**
+     * Creates a Kafka sender that appends records to Kafka topic partitions.
+     *
+     * @param factory A custom producer factory other than the default.
+     * @param options Configuration options of this sender. Changes made to the options
+     *        after the sender is created will not be used by the sender.
+     * @return new instance of Kafka sender
+     */
+    static <K, V> KafkaSender<K, V> create(ProducerFactory factory, SenderOptions<K, V> options) {
```

Comment:
guess it should be public

Reply:
same as above

Reply:
should be public, or at least in java 8. have they changed the rules in java 9 or later?

Reply:
oh, my bad. I thought it is class